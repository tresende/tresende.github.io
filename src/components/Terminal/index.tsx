const Terminal = () => (
  <div className="mb-4 whitespace-nowrap">
    <div className="mb-4 whitespace-nowrap text-greenyellow">
      ~/Projects/tresende.github.io <span className="text-gray-500">main*</span>
    </div>
    <span className="text-cyan-300">❯ </span>
    <span>cat readme.txt</span>
    <span className="blink">_</span>
  </div>
)

export default Terminal
