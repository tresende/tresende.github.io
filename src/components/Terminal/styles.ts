import styled from 'styled-components'

export const InlineText = styled.div`
  margin-bottom: 1rem;
  white-space: nowrap;
`

export const Wrapper = styled(InlineText)``

export const Path = styled(InlineText)`
  color: greenyellow;
`

export const Branch = styled.span`
  color: gray;
`

export const Cursor = styled.span`
  color: aqua;
`

export const Block = styled.span`
  animation-name: blink;
  animation-duration: 700ms;
  animation-iteration-count: infinite;

  @keyframes blink {
    0% {
      opacity: 1;
    }

    50% {
      opacity: 0;
    }

    100% {
      opacity: 1;
    }
  }
`
