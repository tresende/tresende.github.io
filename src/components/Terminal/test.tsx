import { render, screen } from '@testing-library/react'

import Terminal from '.'

describe('<Terminal />', () => {
  it('should start in branch main', () => {
    const { container } = render(<Terminal />)

    expect(screen.getByText(/main/i)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should view readme.txt', () => {
    render(<Terminal />)

    expect(screen.getByText(/cat readme.txt/i)).toBeInTheDocument()
  })
})
