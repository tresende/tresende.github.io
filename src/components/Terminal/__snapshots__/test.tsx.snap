// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Terminal /> should start in branch main 1`] = `
.c0 {
  margin-bottom: 1rem;
  white-space: nowrap;
}

.c1 {
  color: greenyellow;
}

.c2 {
  color: gray;
}

.c3 {
  color: aqua;
}

.c4 {
  -webkit-animation-name: blink;
  animation-name: blink;
  -webkit-animation-duration: 700ms;
  animation-duration: 700ms;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c0 c1"
    >
      ~/Projects/tresende.github.io 
      <span
        class="c2"
      >
        main*
      </span>
    </div>
    <span
      class="c3"
    >
      ‚ùØ 
    </span>
    <span>
      cat readme.txt
    </span>
    <span
      class="c4"
    >
      _
    </span>
  </div>
</div>
`;
